Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,6 +386,22 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        data_x = self.data[:, self.attr_x]
+        data_y = self.data[:, self.attr_y]
+        if self.attr_x not in self.attr_nans:
+            self.attr_nans[self.attr_x] = BasicStats(data_x, self.attr_x).nans
+        if self.attr_y not in self.attr_nans:
+            self.attr_nans[self.attr_y] = BasicStats(data_y, self.attr_y).nans
+        if (self.attr_nans[self.attr_x] == len(data_x) or
+                self.attr_nans[self.attr_y] == len(data_y)):
+        	self.information()
```

Comment:
Yes, this could work (although not indented correctly). You could also clear both, information and warning, unconditionally above this `if` block. Improves readability (imho) and saves three lines. :smiley: